internal/hcs/process.go

```diff
@@ -72,13 +72,45 @@ func (process *Process) SystemID() string {
 	return process.system.ID()
 }
 
+func (process *Process) logGetCommonFields(operation string) logrus.Fields {
+	fields := logrus.Fields{
+		logfields.HCSOperation: operation,
+		logfields.ContainerID:  process.SystemID(),
+		logfields.ProcessID:    process.Pid(),
+	}
+	return fields
+}
+
+func (process *Process) logOperationBegin(operation string) {
+	logOperationBegin(
+		process.logGetCommonFields(operation),
+		"hcsshim::Process - Begin Operation")
+}
+
+func (process *Process) logOperationEnd(operation string, err error) {
+	var result string
+	if err == nil {
+		result = "Success"
+	} else {
+		result = "Error"
+	}
+
+	logOperationEnd(
+		process.logGetCommonFields(operation),
+		"hcsshim::Process - End Operation - "+result,
+		err)
+}
+
 // Signal signals the process with `options`.
-func (process *Process) Signal(options guestrequest.SignalProcessOptions) error {
+func (process *Process) Signal(options guestrequest.SignalProcessOptions) (err error) {
 	process.handleLock.RLock()
 	defer process.handleLock.RUnlock()
-	operation := "Signal"
-	title := "hcsshim::Process::" + operation
-	logrus.Debugf(title+" processid=%d", process.processID)
+
+	operation := "hcsshim::Process::Signal"
+	process.logOperationBegin(operation)
+	defer func() {
```

Comment:
This can be written as `defer process.logOperationEnd(operation, err)`.

You don't need the separate `func()`.

Reply:
Unless this changes the capture rules for `err`?

Reply:
Yes good point. I originally put the error handler in there but now its not needed.